File: Microsoft.AspNet.SignalR.Client.Net40.Samples/CommonClient.cs
using System;
using System.IO;
using System.Threading.Tasks;
using Microsoft.AspNet.SignalR.Client.Hubs;

namespace Microsoft.AspNet.SignalR.Client.Samples
{
    public class CommonClient
    {
        private TextWriter _traceWriter;

        public CommonClient(TextWriter traceWriter)
        {
            _traceWriter = traceWriter;
        }

        public void Run()
        {
            RunHubConnectionAPI();
        }

        private void RunHubConnectionAPI()
        {
            // Url can't be localhost because Windows Phone emulator runs in a separate virtual machine. Therefore, server is located
            // in another machine
            string url = "http://signalr01.cloudapp.net/";

            var hubConnection = new HubConnection(url);
            hubConnection.TraceWriter = _traceWriter;

            var hubProxy = hubConnection.CreateHubProxy("HubConnectionAPI");
            hubProxy.On<string>("displayMessage", (data) => hubConnection.TraceWriter.WriteLine(data));

            hubConnection.Start().Wait();
            hubConnection.TraceWriter.WriteLine("transport.Name={0}", hubConnection.Transport.Name);

            hubProxy.Invoke("DisplayMessageCaller", "Hello Caller!").Wait();

            string joinGroupResponse = hubProxy.Invoke<string>("JoinGroup", hubConnection.ConnectionId, "CommonClientGroup").Result;
            hubConnection.TraceWriter.WriteLine("joinGroupResponse={0}", joinGroupResponse);
            
            hubProxy.Invoke("DisplayMessageGroup", "CommonClientGroup", "Hello Group Members!").Wait();
        }

        private void RunRawConnection()
        {
            string url = "http://signalr01.cloudapp.net/raw-connection";

            var connection = new Connection(url);
            connection.TraceWriter = _traceWriter;

            connection.Start().Wait();
            connection.TraceWriter.WriteLine("transport.Name={0}", connection.Transport.Name);

            connection.Send(new { type = 1, value = "first message" }).Wait();
            connection.Send(new { type = 1, value = "second message" }).Wait();
        }


        private void RunStreaming()
        {
            string url = "http://signalr01.cloudapp.net/streaming-connection";

            var connection = new Connection(url);
            connection.TraceWriter = _traceWriter;

            connection.Start().Wait();
            connection.TraceWriter.WriteLine("transport.Name={0}", connection.Transport.Name);
        }
    }    
}

File: samples/Microsoft.AspNet.SignalR.Client.WebJavaScript.Samples/CommonClient.js
$(new function () {
    $.connection.hubConnectionAPI.client.displayMessage = function (data) {
        log(data);
    }

    $.connection.hub.url = "http://signalr-test1.cloudapp.net:81/signalr";
    $.connection.hub.logging = true;
    $.connection.hub.connectionSlow(function () {
        log("[connectionSlow]");
    });
    $.connection.hub.disconnected(function () {
        log("[disconnected]");
    });
    $.connection.hub.error(function (error) {
        log("[error]" + error);
    });
    $.connection.hub.received(function (payload) {
        log("[received]" + window.JSON.stringify(payload));
    });
    $.connection.hub.reconnected(function () {
        log("[reconnected]");
    });
    $.connection.hub.reconnecting(function () {
        log("[reconnecting]");
    });
    $.connection.hub.starting(function () {
        log("[starting]");
    });
    $.connection.hub.stateChanged(function (change) {
        log("[stateChanged] " + displayState(change.oldState) + " => " + displayState(change.newState));
    });

    $.connection.hub.start().
        done(function () {
            log("Connected");
            log("transport.name=" + $.connection.hub.transport.name);

            $.connection.hubConnectionAPI.server.displayMessageCaller("Hello Caller!");
            $.connection.hubConnectionAPI.server.joinGroup($.connection.hub.id, "CommonClientGroup");
            $.connection.hubConnectionAPI.server.displayMessageGroup("CommonClientGroup", "Hello Group Members!");
        }).
        fail(function (error) {
            log("Failed to connect: " + error);
        });
});

function displayState(state) {
    return ["connecting", "connected", "reconnecting", state, "disconnected"][state];
}

function log(message) {
    $("#Messages").append("<li>[" + new Date().toTimeString() + "] " + message + "</li>");
}